pip/index.py

```diff
@@ -413,13 +419,9 @@ def _find_all_versions(self, project_name):
         for location in url_locations:
             logger.debug('* %s', location)
 
-        formats = set(["source"])
-        if self.use_wheel:
-            formats.add("binary")
-        search = Search(
-            project_name.lower(),
-            pkg_resources.safe_name(project_name).lower(),
-            frozenset(formats))
+        canonical_name = pkg_resources.safe_name(project_name).lower()
+        formats = fmt_ctl_formats(self.format_control, canonical_name)
+        search = Search(project_name.lower(), canonical_name, formats)
```

Comment:
It didnt catch it on the previous PR but it seems strange to set `project_name.lower()` as the user supplied name (since the user obviously supplied `project_name`).